File: Patterns/8. Inverted Pyramid Patterns/77. Inverted Pyramid Binary/ipb.java
import java.util.*;
public class ipb{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        for(int i=1; i<=n; i++){
            // print spaces
            for(int j=1; j<=i-1; j++){
                System.out.print("  ");
            }
            // print first half
            for(int j=1; j<=n-i+1; j++){
                System.out.print(j%2 + " ");
            }
            // print second half
            for(int j=n-i; j>=1; j--){
                System.out.print(j%2 + " ");
            }
            System.out.println();
        }
    }
}

/*
Output

1 0 1 0 1 0 1 0 1
  1 0 1 0 1 0 1
    1 0 1 0 1
      1 0 1
        1 

*/

File: Patterns/08. Inverted Pyramid Patterns/77. Inverted Pyramid Binary/ipb.py
n = int(input("Enter a Number : "))

for i in range(1,n+1):
    # print spaces
    for j in range(1,i):
        print(end='  ')
    # print first half
    for j in range(1,n-i+2):
        print(j%2,end=' ')
    # print second half
    for j in range(n-i,0,-1):
        print(j%2,end=' ')
    print()


'''

Output:

1 0 1 0 1 0 1 0 1 
  1 0 1 0 1 0 1 
    1 0 1 0 1 
      1 0 1 
        1  

'''